packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -27,67 +33,161 @@ import {
   getTruncateNumbersWithPound,
   getSettings,
   saveSettings as saveSettingsAction,
+  RootState,
+  WorkspaceSettings,
 } from '@deephaven/redux';
-import { DbNameValidator, TimeUtils } from '@deephaven/utils';
+import { assertNotNull, DbNameValidator, TimeUtils } from '@deephaven/utils';
 import './FormattingSectionContent.scss';
+import type { DebouncedFunc } from 'lodash';
 
 const log = Log.module('FormattingSectionContent');
 
-export class FormattingSectionContent extends PureComponent {
+type FormatterItem = {
+  columnType: string;
+  columnName: string;
+  format: Partial<TableColumnFormat>;
+  id?: number;
+  isNewRule?: boolean;
+};
+
+type FormatOption = {
+  defaultFormatString?: string;
+};
+
+interface FormattingSectionContentProps {
+  formatter?: FormatterItem[];
```

Comment:
`formatter` should be required. There's a bunch of props that used to be required that you've marked optional here, such as `defaultDateTimeFormat`, `showTimeZone`, `showTSeparator`, etc:
```
FormattingSectionContent.propTypes = {
  formatter: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  defaultDateTimeFormat: PropTypes.string.isRequired,
  showTimeZone: PropTypes.bool.isRequired,
  showTSeparator: PropTypes.bool.isRequired,
  timeZone: PropTypes.string.isRequired,
  truncateNumbersWithPound: PropTypes.bool.isRequired,
  settings: PropTypes.shape({}).isRequired,
  saveSettings: PropTypes.func.isRequired,
  scrollTo: PropTypes.func,
  defaultDecimalFormatOptions: PropTypes.shape({
    defaultFormatString: PropTypes.string,
  }).isRequired,
  defaultIntegerFormatOptions: PropTypes.shape({
    defaultFormatString: PropTypes.string,
  }).isRequired,
  defaults: PropTypes.shape({
    defaultDateTimeFormat: PropTypes.string.isRequired,
    defaultDecimalFormatOptions: PropTypes.shape({
      defaultFormatString: PropTypes.string,
    }).isRequired,
    defaultIntegerFormatOptions: PropTypes.shape({
      defaultFormatString: PropTypes.string,
    }).isRequired,
    showTimeZone: PropTypes.bool.isRequired,
    showTSeparator: PropTypes.bool.isRequired,
    timeZone: PropTypes.string.isRequired,
  }),
};
```

Reply:
The `FormattingSectionContent.tsx` is used in `SettingsMenu.tsx` with only one prop passed in, which is `scrollTo`. Most of these props also don't have default values. This would mean the props are optional?